plugin/db/mysql/mysql.go

```diff
@@ -938,6 +950,104 @@ func (driver *Driver) Restore(ctx context.Context, sc *bufio.Scanner) (err error
 	return nil
 }
 
+// Ghost table names are generated by appending "_ghost" to the original ones.
+// Constraint names are generated by appending a timestamp (to second) to the original ones.
+// According to https://dev.mysql.com/doc/refman/8.0/en/identifier-length.html, the max length of a
+// constraint name is 64, so we should truncate the original constraint name part to fit in 64 characters.
+func rewriteToGhostTableForRestore(stmt, timestamp string) string {
+	reInsert := regexp.MustCompile(`INSERT INTO ` +
+		// Capture group for optional db prefix like "`dbPrefix`."
+		// the ?: means DO NOT capture this group
+		// the last ? means the whole db prefix is optional
+		"(?:`([^`]+)`\\.)?" +
+		// Capture group for the table name, leaving back quote ` outside for convenience
+		"`([^`]+)` " +
+		// Capture group for values to the end of the line
+		"VALUES (.*)")
+
+	reCreate := regexp.MustCompile("CREATE TABLE " +
+		// Capture group for the table name, leaving back quote ` outside for convenience
+		"`([^`]+)` " +
+		// Capture group for the rest of the contents
+		// Note that there will be multiple lines, so using [\s\S]+ to match them all including line breaks
+		`([\s\S]+)`)
+
+	// matchesInsert would be like:
+	// with dbPrefix: []string{"INSERT INTO `dbPrefix`.`table` VALUES (1, 2, 3);", "dbPrefix", "table", "(1, 2, 3);"}
+	// empty dbPrefix: []string{"INSERT INTO `table` VALUES (1, 2, 3);", "", "table", "(1, 2, 3);"}
+	matchesInsert := reInsert.FindStringSubmatch(stmt)
+
+	// matchesCreate would be like:
+	// []string{"CREATE TABLE `table` (", "table", "("}
+	matchesCreate := reCreate.FindStringSubmatch(stmt)
+
+	var stmtWithGhostTables string
+	if matchesInsert != nil {
+		dbPrefix := ""
```

Comment:
I don't think we need to worry about the dbPrefix case because a single database dump should not have a prefix.

Or, we can return an error if it has a prefix.

Reply:
I prefer to return an error if it has a db prefix, which implies unexpected conditions.